# 0x05. Python - Exceptions

## General

- Why Python programming is awesome
- What’s the difference between errors and exceptions
- What are exceptions and how to use them
- When do we need to use exceptionss
- How to correctly handle an exception
- What’s the purpose of catching exceptions
- How to raise a builtin exception
- When do we need to implement a clean-up action after an exception

| s/n   | Description                                                             | name                          |
| ----- | ----------------------------------------------------------------------- | ----------------------------- |
| - [x] | A function that prints x elements of a list                             | 0-safe_print_list.py          |
| - [x] | A function that prints an integer with "{:d}".format()                  | 1-safe_print_integer.py       |
| - [x] | A function that prints the first x elements of a list and only integers | 2-safe_print_list_integers.py |
| - []  | A function that returns a set of common elements in two sets            | 3-safe_print_division.py      |
| - []  | A function that returns a set of all elements present in only one set   | 4-list_division.py            |
| - [x] | A function that raises a type exception                                 | 5-raise_exception.py          |
| - [x] | A function that raises a name exception with a message                  | 6-raise_exception_msg.py      |

:sparkles:
